metrics_sdk/test/opentelemetry/sdk/metrics/aggregation/explicit_bucket_histogram_test.rb

```diff
@@ -0,0 +1,262 @@
+      _(hdps[1].max).must_equal(80)
+      _(hdps[1].bucket_counts).must_equal([1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0])
+      _(hdps[1].explicit_bounds).must_equal([0, 5, 10, 25, 50, 75, 100, 250, 500, 1000])
+    end
+
+    it 'sets the timestamps' do
+      ebh.update(0, {})
+      hdp = ebh.collect(now_in_nano, now_in_nano)[0]
+      _(hdp.start_time_unix_nano).must_equal(now_in_nano)
+      _(hdp.time_unix_nano).must_equal(now_in_nano)
+    end
+
+    it 'calculates the bucket count' do
+      ebh.update(0, {})
+      ebh.update(0, {})
+      ebh.update(0, {})
+      ebh.update(0, {})
+      hdp = ebh.collect(now_in_nano, now_in_nano)[0]
+      _(hdp.count).must_equal(4)
+    end
+
+    it 'does not aggregate between collects' do
+      ebh.update(0, {})
+      ebh.update(1, {})
+      ebh.update(5, {})
+      ebh.update(6, {})
+      ebh.update(10, {})
+      hdps = ebh.collect(now_in_nano, now_in_nano)
+
+      ebh.update(0, {})
+      ebh.update(1, {})
+      ebh.update(5, {})
+      ebh.update(6, {})
+      ebh.update(10, {})
+      # Assert that the recent update does not
+      # impact the already collected metrics
+      _(hdps[0].count).must_equal(5)
+      _(hdps[0].sum).must_equal(22)
+      _(hdps[0].min).must_equal(0)
+      _(hdps[0].max).must_equal(10)
+      _(hdps[0].bucket_counts).must_equal([1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0])
+
+      hdps = ebh.collect(now_in_nano, now_in_nano)
+      # Assert that we are not accumulating values
+      # between calls to collect
+      _(hdps[0].count).must_equal(5)
+      _(hdps[0].sum).must_equal(22)
+      _(hdps[0].min).must_equal(0)
+      _(hdps[0].max).must_equal(10)
+      _(hdps[0].bucket_counts).must_equal([1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0])
+    end
+
+    describe 'when aggregation_temporality is not delta' do
+      let(:aggregation_temporality) { :not_delta }
+
+      it 'allows metrics to accumulate' do
+        ebh.update(0, {})
+        ebh.update(1, {})
+        ebh.update(5, {})
+        ebh.update(6, {})
+        ebh.update(10, {})
+        hdps = ebh.collect(now_in_nano, now_in_nano)
+
+        ebh.update(0, {})
+        ebh.update(1, {})
+        ebh.update(5, {})
+        ebh.update(6, {})
+        ebh.update(10, {})
+        # Assert that the recent update does not
+        # impact the already collected metrics
+        _(hdps[0].count).must_equal(5)
+        _(hdps[0].sum).must_equal(22)
+        _(hdps[0].min).must_equal(0)
+        _(hdps[0].max).must_equal(10)
+        _(hdps[0].bucket_counts).must_equal([1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0])
+
+        hdps = ebh.collect(now_in_nano, now_in_nano)
+        # Assert that we are accumulating values
+        # and not just capturing the delta since
+        # the previous collect call
+        _(hdps[0].count).must_equal(10)
+        _(hdps[0].sum).must_equal(44)
+        _(hdps[0].min).must_equal(0)
+        _(hdps[0].max).must_equal(10)
+        _(hdps[0].bucket_counts).must_equal([2, 4, 4, 0, 0, 0, 0, 0, 0, 0, 0])
```

Comment:
I'd like us to also assert that the previously collected metrics were not modified by the most recent `collect` call.
```suggestion
        hdps1 = ebh.collect(now_in_nano, now_in_nano)
        # Assert that we are accumulating values
        # and not just capturing the delta since
        # the previous collect call
        _(hdps1[0].count).must_equal(10)
        _(hdps1[0].sum).must_equal(44)
        _(hdps1[0].min).must_equal(0)
        _(hdps1[0].max).must_equal(10)
        _(hdps1[0].bucket_counts).must_equal([2, 4, 4, 0, 0, 0, 0, 0, 0, 0, 0])

        # Assert that the recent collect does not
        # impact the already collected metrics
        _(hdps[0].count).must_equal(5)
        _(hdps[0].sum).must_equal(22)
        _(hdps[0].min).must_equal(0)
        _(hdps[0].max).must_equal(10)
        _(hdps[0].bucket_counts).must_equal([1, 2, 2, 0, 0, 0, 0, 0, 0, 0, 0])
```